csss-site/src/tools/lib/generate_document.py

```diff
@@ -0,0 +1,222 @@
+import typing
+import io
+import os
+
+from dataclasses import dataclass
+from datetime import datetime
+
+from pypdf import PdfReader, PdfWriter
+from pypdf.generic import NameObject
+from reportlab.pdfgen import canvas
+
+# definitions:
+# - Cheque Requisison (or Cheque-Req for short) is a document that requests money from the sfss
+# -
+
+# ------------------------
+# globals
+
+# https://sfss.ca/wp-content/uploads/2022/09/Cheque-Requisition-Form-Pick-Up-2022-1P.pdf
+small_text_logo = "tools/static/logo-text-smaller.png"
+cheque_req_form_pickup = "tools/static/Cheque-Requisition-Form-Pick-Up-2022-1P.pdf"
+cheque_req_form_mailed = ""
+
+tmp_dir = "tools/static/tmp/"
+
+
+# ------------------------
+# structs
+
+# NOTE: these are pretty much just C-style structs
+# TODO: have example values for all dataclasses
+
+@dataclass
+class Pickup:
+    pickup_by: str
+    pickup_email: str
+
+
+@dataclass
+class Mailed:
+    mail_off_campus: bool
+    street_address: str
+    city_province: str
+    postal_code: str
+
+
+@dataclass
+class CoreChequeReq:
+    request_by: str
+    requester_position: str
+    request_desc: str
+
+    payable_to: str
+    amount_cad: float
+    receive_kind: typing.Union[Pickup, Mailed]
+
+    current_date: datetime
+
+
+@dataclass
+class ReceiptInfo:
+    paid_by: str
+    location: str
+    amount_cad: float
+
+# ------------------------
+# functions
+
+
+# generate cheque req from core
+# meeting_minutes: str is the prefix for the meeting minutes of the form: "yyyy-mm-dd"
+def generate_core_cheque_req(
+    data: CoreChequeReq,
+        meeting_minutes: str,
+        receipts: list[ReceiptInfo],
+        include_auto_desc: bool = True,
+        watermark_active: bool = True):
+
+    reader = PdfReader(cheque_req_form_pickup)
+    assert len(reader.pages) == 1

```

Comment:
don't use asserts, Return an error message back to the user that can try to explain what went wrong, you can look for example of how to do so in the `elections` app, look for `return False` to see how it's handled.

Reply:
Good point! This assert doesn't actually depend on user input, but there should definitely be some form of description so debugging is easier if a developer makes a breaking change.